lib/hyperion/headers.rb

```diff
@@ -1,23 +1,43 @@
+require 'hyperion/formats'
+
 class Hyperion
   module Headers
+    # constructs and destructures HTTP headers
 
-    def post_headers(format)
-      {
-          'Content-Type' => content_type(format)
-      }
+    def route_headers(route)
+      headers = {}
+      rd = route.response_descriptor
+      pd = route.payload_descriptor
+      headers['Expect'] = 'x' # this overrides default libcurl behavior.
+                              # see http://devblog.songkick.com/2012/11/27/a-second-here-a-second-there/
+                              # the value has to be non-empty or else it is ignored
+      if rd
+        headers['Accept'] = "application/vnd.#{Hyperion.config.vendor_string}.#{short_mimetype(rd)}"
+      end
+      if pd
+        headers['Content-Type'] = content_type_for(pd.format)
+      end
+      headers
     end
 
-    def default_headers(type, version, format)
-      {
-          'Accept' => "application/vnd.indigobio-ascent.#{type}-v#{version}+#{format}"
-      }
+    def short_mimetype(response_descriptor)
+      x = response_descriptor
+      "#{x.type}-v#{x.version}+#{x.format}"
     end
 
-    def content_type(format)
-      case format
-        when 'json'; 'application/json'
-      end
+    ContentTypes = [[:json, 'application/json'],
+                    [:protobuf, 'application/x-protobuf']]
+
```

Comment:
Good point. That's a task for a later date. I've added a note to that effect in the README.